### ISSUE TITLE
[kots] Allow adding additional credentials for private registries in KOTS

### ISSUE THREAD

**Issue by corneliusludmann** [gitpod-io/gitpod#12136]
In the KOTS UI, the only credentials for registries a user can provide are for the external registry where the workspace images are pushed to. To be able to pull workspace base images from a different private registry users should have the option to add additional registry credentials.

This is relevant for setups that are using the built-in registry as well as an external registry.

When using the built-in registry, for a proper solution these issues need to be solved first:
- #10792
- #12060

However, **to unblock current customers**, let's implement a similar hack as [we did for air-gap setups](https://github.com/gitpod-io/gitpod/pull/10685/files#diff-633cc9aa71daacd7993fd8af48e3d65d636b1e9a895f0752ff59a98d8c929766R307-R308). Once the above issues are solved, we can change this implementation.

There are 2 options to implement this:
1. Let users upload a `.dockerconfig` file in the KOTS UI
2. Let users specify a secret name where they previously uploaded a `.dockerconfig`.

In the installer job, we would read the `.dockerconfig` file (that the user uploaded or from the secret) and merge it with the built-in registry secret or the given username / password of the external registry.

Implementing (2) (user specifies a secret name) has the drawback that changing the secret content does not have any effect until the installer job is running again (because the installer job merges the secret content with the built-in / new created secret). Also, (1) (user uploads a `.dockerconfig` file) fits more what we are currently doing in the KOTS UI (specifying credentials and let KOTS create the secrets). That's why I would vote for (1) for now.

In the long term, it would be great if users can choose whether they want to provide credentials via secrets or directly in the KOTS UI. However, the UI features of KOTS are pretty limited.

To summarize the to-do's of this issue:
- Add a field in the KOTS UI to upload a `.dockerconfig` file with credentials to private registries, no matter if the built-in or an external registry is used.
- In the installer job, merge all registry credentials (built-in or external registry, KOTS air-gap registry, additional credentials from the uploaded `.dockerconfig`) to a common `.dockerconfig` that a) replaces the built-in registry secret (in case the built-in registry has been chosen) or b) is used to create a new registry credentials secret that is passed to the installer as external registry credentials.

**Comment by Pothulapati** [comment]
@corneliusludmann This makes sense, and provides us a way to fix it without having to wait on https://github.com/gitpod-io/gitpod/issues/12060
